### ISSUE TITLE
Trying to read past the end of the stream

### ISSUE THREAD

**Issue by dangmai** [saul/demofile#14]
I'm trying to use demofile to parse a recently played demo file, and it's failing right after the match starts (This happens to multiple demo files). Here's the output from running the `examples/dumpfile.js`:

```
Demo header: { magic: 'HL2DEMO',
  protocol: 4,
  networkProtocol: 13569,
  serverName: 'Valve CS:GO US SouthEast Server (srcds007.125.44)',
  clientName: 'GOTV Demo',
  mapName: 'de_cache',
  gameDirectory: 'csgo',
  playbackTime: 3098.859375,
  playbackTicks: 198327,
  playbackFrames: 99125,
  signonLength: 614211 }
cash_team_rescued_hostage: undefined -> 600
bot_autodifficulty_threshold_high: undefined -> 0
cash_team_win_by_defusing_bomb: undefined -> 3500
game_mode: undefined -> 1
mp_ggtr_bomb_pts_for_upgrade: undefined -> 2
ff_damage_reduction_bullets: undefined -> 0.33
bot_quota_mode: undefined -> fill
cash_player_interact_with_hostage: undefined -> 300
mp_maxrounds: undefined -> 30
ammo_grenade_limit_total: undefined -> 4
mp_respawn_immunitytime: undefined -> 0
mp_roundtime_defuse: undefined -> 1.92
mp_ggprogressive_round_restart_delay: undefined -> 15
mp_timelimit: undefined -> 0
mp_warmuptime: undefined -> 60
sv_deadtalk: undefined -> 1
mp_randomspawn_los: undefined -> 0
mp_weapons_allow_map_placed: undefined -> 1
sv_hosting_lobby: undefined -> 1
nextlevel: undefined -> de_cache
mp_ggtr_bomb_defuse_bonus: undefined -> 1
mp_buytime: undefined -> 20
bot_difficulty: undefined -> 2
mp_roundtime_hostage: undefined -> 1.92
mp_freezetime: undefined -> 15
bot_defer_to_human_goals: undefined -> 1
sv_skyname: undefined -> sky_cs15_daylight01_hdr
sv_reliableavatardata: undefined -> 1
mp_friendlyfire: undefined -> 1
cash_team_hostage_interaction: undefined -> 600
spec_freeze_panel_extended_time: undefined -> 0
ff_damage_reduction_other: undefined -> 0.4
mp_molotovusedelay: undefined -> 0
cash_team_elimination_hostage_map_t: undefined -> 3000
mp_ggtr_bomb_detonation_bonus: undefined -> 1
ammo_grenade_limit_flashbang: undefined -> 2
mp_roundtime: undefined -> 1.92
mp_halftime: undefined -> 1
mp_ggtr_bomb_respawn_delay: undefined -> 0
think_limit: undefined -> 0
ff_damage_reduction_grenade: undefined -> 0.85
cash_team_elimination_hostage_map_ct: undefined -> 3000
cash_team_win_by_hostage_rescue: undefined -> 2900
tv_transmitall: undefined -> 1
GOTV (BOT) joined the game
syncopate (STEAM_1:1:18164814) joined the game
Larry (BOT) joined the game
Bert (BOT) joined the game
*** Round ended 'first' (reason: 16)
	Terrorists:  score 0
	CTs:  score 0
Fergus (BOT) joined the game
Bob (BOT) joined the game
Ryan (BOT) joined the game
Yanni (BOT) joined the game
Graham (BOT) joined the game
Allen (BOT) joined the game
Opie (BOT) joined the game
TAK1N L1VES (STEAM_1:0:42267311) joined the game
/home/dangmai/Dev/demo-analyzer/node_modules/demofile/ext/bitbuffer.js:201
  CW_LowPrecision,
           ^

Error: Trying to read past the end of the stream
    at BitStream.<anonymous> (/home/dangmai/Dev/demo-analyzer/node_modules/bit-buffer/bit-buffer.js:191:10)
    at _.map (/home/dangmai/Dev/demo-analyzer/node_modules/demofile/ext/bitbuffer.js:46:54)
    at arrayMap (/home/dangmai/Dev/demo-analyzer/node_modules/demofile/node_modules/lodash/index.js:1406:25)
    at Function.map (/home/dangmai/Dev/demo-analyzer/node_modules/demofile/node_modules/lodash/index.js:6710:14)
    at BitStream.bitBuffer.BitStream.readBytes (/home/dangmai/Dev/demo-analyzer/node_modules/demofile/ext/bitbuffer.js:46:23)
    at _.reduce (/home/dangmai/Dev/demo-analyzer/node_modules/demofile/stringtables.js:217:34)
    at arrayReduce (/home/dangmai/Dev/demo-analyzer/node_modules/demofile/node_modules/lodash/index.js:1450:23)
    at Function.<anonymous> (/home/dangmai/Dev/demo-analyzer/node_modules/demofile/node_modules/lodash/index.js:3445:13)
    at StringTables._parseStringTableUpdate (/home/dangmai/Dev/demo-analyzer/node_modules/demofile/stringtables.js:175:7)
    at StringTables._handleUpdateStringTable (/home/dangmai/Dev/demo-analyzer/node_modules/demofile/stringtables.js:287:10)
```

**Comment by dangmai** [comment]
Downgrading to 0.4.1 fixes this issue, it seems to me that it's a problem with bit-buffer 0.1.0

**Comment by saul** [comment]
Thanks, fixed in 0.4.3 Was not supposed to upgrade bit-buffer.
